jxl/src/container/box_header.rs

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) the JPEG XL Project Authors. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+//
+// Originally written for jxl-oxide.
+
+use crate::error::Error;
+
+/// Box header used in JPEG XL containers.
+#[derive(Debug, Clone)]
+pub struct ContainerBoxHeader {
+    ty: ContainerBoxType,
+    box_size: Option<u64>,
+    is_last: bool,
+}
+
+pub enum HeaderParseResult {
+    Done {
+        header: ContainerBoxHeader,
+        header_size: usize,
+    },
+    NeedMoreData,
+}
+
+impl ContainerBoxHeader {
+    pub(super) fn parse(reader: &super::ConcatSlice<'_, '_>) -> Result<HeaderParseResult, Error> {
+        let mut buf = [0u8; 16];
+        let buf = reader.peek(&mut buf);
+        if buf.len() < 8 {
```

Comment:
Doesn't this follow anyway from the default case in the match?

Reply:
Ooh, right -- I've removed the if statement.